### ISSUE TITLE
Counterpart for fakefullscreen

### ISSUE THREAD

**Issue by JustSimplyKyle** [hyprwm/Hyprland#1817]
FakeFullscreen would currently send the fullscreen state but not alter the geometry.
What I want is I want to alter the geometry to be fullscreened, but not the state.
The main usage for this would be fullscreening a wayland chromium without losing the title bar

tl;dr fullscreen the app without telling it

**Comment by devadathanmb** [comment]
Hey, Is this being worked on? Would love to see if this feature gets merged in.

**Comment by vaxerski** [comment]
No, it's not. I have a kjhillion other things to work on. If you really want it that bad, you can always make a pr.

If you can't / don't want to, you can like the original post instead of posting useless comments

**Comment by GrabbenD** [comment]
Thanks a lot for implementing this @vaxerski!

@JustSimplyKyle 
```
# Exclusive fullscreen with mainMod + P
bind = $mainMod, F, fullscreen, 0

# Windowed fullscreen with mainMod + SHIFT + P
bind = $mainMod SHIFT, F, fullscreen, 1
```

**Comment by vaxerski** [comment]
wat

**Comment by GrabbenD** [comment]
@vaxerski I was skimming through the documentation and noticed `fullscreen, 0` (takes your entire screen) and `fullscreen, 1` (keeps gaps and bar(s)). Mode 1 is exactly what I was looking for in #1923 and I suppose that's what @JustSimplyKyle requested in this PR:
> The main usage for this would be fullscreening a wayland chromium without losing the title bar

Sounds like this PR is resolved unless I've misunderstood something :)

**Comment by vaxerski** [comment]
you have misunderstood it

**Comment by GrabbenD** [comment]
I suppose the idea is to just keep bars but not gaps in that case?

**Comment by vaxerski** [comment]
pleease stop trashing this conversation

**Comment by fraschm1998** [comment]
> @vaxerski I was skimming through the documentation and noticed `fullscreen, 0` (takes your entire screen) and `fullscreen, 1` (keeps gaps and bar(s)). Mode 1 is exactly what I was looking for in #1923 and I suppose that's what @JustSimplyKyle requested in this PR:
> 
> > The main usage for this would be fullscreening a wayland chromium without losing the title bar
> 
> Sounds like this PR is resolved unless I've misunderstood something :)

I think he's referring to something like the dwm patch https://dwm.suckless.org/patches/fakefullscreen/

**Comment by AIndoria** [comment]
> pleease stop trashing this conversation

What an absolutely weird thing to say over a question.

**Comment by vaxerski** [comment]
I don't know on which forum post this message got linked but cherry-picking one message of mine when I was probably having a shit day half a year ago is also weird, imo.

**Comment by Faez-Ansari** [comment]
When I use fullscreen 0, it fullscreens and send fullscreen signal to browser. If I set it to one it doesn't even fullscreen anymore

**Comment by LucStr** [comment]
Hello everyone. I've been having the exact same issue as described in here. For my workflow I like to work with fullscreen, but some applications mainly chromium think it's a good idea to just go into presentation mode upon being prompted to go to fullscreen mode. Additionally jetbrains IDEs also encountered an issue where it automatically leaves fullscreen if you go fullscreen https://youtrack.jetbrains.com/issue/IDEA-330605/IntelliJ-automatically-leaving-full-screen-in-tiling-WM 

I am not a c++ developer by any means, but I tried my best in debugging my issue and getting a workaround for it. It is actually fairly simple in my case https://github.com/hyprwm/Hyprland/commit/4afae3470198ab1c974e6182c9eec4e91c8ec881 This will completely remove the functionality of using the applications native fullscreen mode. For my usecase this is fine, at least so far. I might encounter issues down the line. 

@vaxerski  I am not sure if I would be able to get this into a state where it would be worth to get merged, but from an implementation perspective I would assume I would create a new value in eFullscreenMode and replace all the references of where FULLSCREEN_FULL is used with || operators to encompass a new mode?

To get nixos to properly apply the patch I've used this https://github.com/LucStr/nix-config/commit/c8748a39c0927249b23314b70b00ec865fdbdfee (I am new to nixos so this was a bit of a struggle for me and might not be best practice)
